dev/preview/workflow/preview/build.sh

```diff
@@ -15,8 +15,20 @@ ensure_gcloud_auth
 
 VERSION="$(preview-name-from-branch)-dev"
 
+# We have quite a few Leeway packages whose hash includes files from .gitignore.
+# Some of these files (such as build folders) are populated as part of our prebuilds
+# which means that the hash for these components will be different from a workspace
+# then in CI.
+#
+# We should fix the package definitions so that they don't include these files in the hash.
+#
+# For now we're simply cleaning the repository by deleting all files that are in .gitignore.
+git clean -dfX
```

Comment:
Let's do this only in werft, as otherwise it might nuke changes that have not been added yet.

```suggestion
if [[ -n ${WERFT_SERVICE_HOST+x} ]]; then
  git clean -dfX
fi
```

Reply:
This is only really needed outside of Werft though - I don't think it will remove changed files (will test), but you're right it would delete files that are not yet tracked by git! Let me see if I can't narrow it down to only cover "ignored files" ☺️